odoo/addons/base/res/res_company.py

```diff
@@ -158,20 +158,23 @@ def _get_user_currency(self):
         ('name_uniq', 'unique (name)', 'The company name must be unique !')
     ]
 
+    def _sync_adress_fields(company, partner):
+        company.street = partner.street
+        company.street2 = partner.street2
+        company.city = partner.city
+        company.zip = partner.zip
+        company.state_id = partner.state_id
+        company.country_id = partner.country_id
+        company.fax = partner.fax
+
     # TODO @api.depends(): currently now way to formulate the dependency on the
     # partner's contact address
     def _compute_address(self):
         for company in self.filtered(lambda company: company.partner_id):
             address_data = company.partner_id.sudo().address_get(adr_pref=['contact'])
             if address_data['contact']:
                 partner = company.partner_id.browse(address_data['contact'])
-                company.street = partner.street
-                company.street2 = partner.street2
-                company.city = partner.city
-                company.zip = partner.zip
-                company.state_id = partner.state_id
-                company.country_id = partner.country_id
-                company.fax = partner.fax
+                self._sync_adress_fields(company, partner)
```

Comment:
Not a bad idea, but if you touch it you should imo also do a proper single write with a dictionary instead of multiple writes.
```python
company.write(self._get_address_fields(partner))
```